Create Rule “avoid-repeated-emojis/rule”

JackDevAU opened the pull request:
Automatically generated by Netlify CMS

tiagov8 commented on rules/avoid-repeated-emojis/rule.md:
```diff
@@ -0,0 +1,63 @@
+Following the DRY principle by avoiding excessive repetition of emojis helps to create lists that are visually interesting and easy to read, while also promoting efficient and maintainable content creation.
+
+
+::: tip
```
Hey @JackDevAU 

There is not box type "tip"
See https://ssw.com.au/rules/rule/#5-boxes 

1. Please fix
I would add just the prefix "Tip:" in bold in front of the sentence

JackDevAU replied:
```suggestion
**Tip:**
```

tiagov8 commented on rules/avoid-repeated-emojis/rule.md:
```diff
@@ -0,0 +1,63 @@
+
+
+::: tip
+When creating a list that includes emojis, avoid repeating the same emoji more than three times within the list. This helps to keep the list concise, readable, and consistent while still allowing for some repetition for emphasis or clarity. 
```
1. Please follow https://ssw.com.au/rules/use-digits-instead-of-words/

JackDevAU replied:
```suggestion
When creating a list that includes emojis, avoid repeating the same emoji more than 3 times within the list. This helps to keep the list concise, readable, and consistent while still allowing for some repetition for emphasis or clarity. 
```

tiagov8 commented on rules/avoid-repeated-emojis/rule.md:
```diff
@@ -0,0 +1,63 @@
+- Can be challenging to maintain
+:::
+::: good
+Good example – using the DRY principle 
```
1. Please make "Using" capitalized for consistency ;)

JackDevAU replied:
```suggestion
Good example – Using the DRY principle 
```

tiagov8 commented on rules/avoid-repeated-emojis/rule.md:
```diff
@@ -0,0 +1,63 @@
+Emojis are a lot easier to consume when a lot of information is present. But when repeated excessively, they can become a hindrance rather than a help. This helps to make it more visually interesting and easier to scan quickly. In this way, emojis can enhance the communication experience, rather than detracting from it.
+
+<!--endintro-->
+
```
Hey @JackDevAU 

1. Can you add a ::: info box here with a short defintion of DRY? E.g. https://www.digitalocean.com/community/tutorials/what-is-dry-development

> D_RY, which stands for ‘don’t repeat yourself,’ is a principle of software development that aims at reducing the repetition of patterns and code duplication in favor of abstractions and avoiding redundancy._

tiagov8 commented:
Hey @JackDevAU 

I was thinking... this rule can also be useful for words not just emojis... 

What do you think about changing from:

```
title: DRY - Do you avoid repeated emojis in lists?
uri: avoid-repeated-emojis
```

to

```
title: DRY - Do you avoid repeated elements in lists?
uri: avoid-repetition
```

**Note:** Require update in the category + folder name as well

---

Then add another bad and good example... e.g.:

```

  This is also valid for words in lists. You should keep only the part that is unique in each list item.

  ::: greybox
  Following this rule is important:
  
  - To help you increase productivity
  - To help you save time
  - To help you reduce stress
 
  :::
  ::: bad
  Bad example – Repeating words... Not following DRY :(
  :::

  ::: greybox
  Following this rule is important to help you:
  
  - Increase productivity
  - Save time
  - Reduce stress
 
  :::
  ::: good
  Good example – No repeated words... using the DRY principle 
  :::
```

JackDevAU commented on rules/avoid-repeated-emojis/rule.md:
```diff
@@ -0,0 +1,63 @@
+
+::: tip
+When creating a list that includes emojis, avoid repeating the same emoji more than three times within the list. This helps to keep the list concise, readable, and consistent while still allowing for some repetition for emphasis or clarity. 
+:::
```
```suggestion

```

JackDevAU commented:
> Hey @JackDevAU
> 
> I was thinking... this rule can also be useful for words not just emojis...
> 
> What do you think about changing from:
> 
> ```
> title: DRY - Do you avoid repeated emojis in lists?
> uri: avoid-repeated-emojis
> ```
> 
> to
> 
> ```
> title: DRY - Do you avoid repeated elements in lists?
> uri: avoid-repetition
> ```
> 
> **Note:** Require update in the category + folder name as well
> 
> Then add another bad and good example... e.g.:
> 
> ```
> 
>   This is also valid for words in lists. You should keep only the part that is unique in each list item.
> 
>   ::: greybox
>   Following this rule is important:
>   
>   - To help you increase productivity
>   - To help you save time
>   - To help you reduce stress
>  
>   :::
>   ::: bad
>   Bad example – Repeating words... Not following DRY :(
>   :::
> 
>   ::: greybox
>   Following this rule is important to help you:
>   
>   - Increase productivity
>   - Save time
>   - Reduce stress
>  
>   :::
>   ::: good
>   Good example – No repeated words... using the DRY principle 
>   :::
> ```

Agreed! I've updated the title and added the example to the rule
